sdk/search/azure-search/src/main/java/com/azure/search/documents/models/PatternAnalyzer.java

```diff
@@ -112,8 +114,9 @@ public RegexFlags getFlags() {
      * @param flags the flags value to set.
      * @return the PatternAnalyzer object itself.
      */
-    public PatternAnalyzer setFlags(RegexFlags flags) {
-        this.flags = flags;
+    public PatternAnalyzer setFlags(Collection<RegexFlags> flags) {
```

Comment:
This API may be better as `setFlags(RegexFlags... flags)`.

Reply:
The reason I choose this because we do not need to convert if passing RegexFlags.values()

Reply:
Would this be a common scenario though? Also, I think this would be likely to fail as `values()` would contain all the unioned `RegexFlags` as well.

Reply:
@brjohnstmsft what does the service do if the same flag is passed multiple times.